wasmbinding/stargate_query_test.go

```diff
@@ -0,0 +1,143 @@
+
+		// ibc transfer
+		"/ibc.applications.transfer.v1.Query/DenomTrace",
+		"/ibc.applications.transfer.v1.Query/Params",
+		"/ibc.applications.transfer.v1.Query/DenomHash",
+		"/ibc.applications.transfer.v1.Query/EscrowAddress",
+		"/ibc.applications.transfer.v1.Query/TotalEscrowForDenom",
+	} {
+		stargateQueryPaths.Add(queryPath)
+	}
+
+	t.Log("stargateQueryPaths: Add nibiru query paths from GRPC service descriptions")
+	queryServiceDescriptions := []grpc.ServiceDesc{
+		epochs.GrpcQueryServiceDesc(),
+		devgas.GrpcQueryServiceDesc(),
+		inflation.GrpcQueryServiceDesc(),
+		oracle.GrpcQueryServiceDesc(),
+		sudotypes.GrpcQueryServiceDesc(),
+		tokenfactory.GrpcQueryServiceDesc(),
+	}
+	for _, serviceDesc := range queryServiceDescriptions {
+		for _, queryMethod := range serviceDesc.Methods {
+			stargateQueryPaths.Add(
+				fmt.Sprintf("/%v/%v", serviceDesc.ServiceName, queryMethod.MethodName),
+			)
+		}
+	}
+
+	gotQueryPaths := []string{}
+	// It's not required for the response type and the method description of the
+	// stargate query's gRPC path to match up exactly as expected. The exception
+	// to this convention is when our response type doesn't stripped of its
+	// "Response" suffix and "Query" prefix is not the same as the method name.
+	// This happens when "QueryAAARequest" does not return a "QueryAAAResponse".
+	exceptionPaths := set.New[string]("/nibiru.oracle.v1.QueryExchangeRateResponse")
+	for queryPath, protobufResponse := range wasmbinding.WasmAcceptedStargateQueries() {
+		gotQueryPaths = append(gotQueryPaths, queryPath)
+
+		// Show that the underlying protobuf name and query paths coincide.
+		pbQueryResponseTypeUrl := "/" + proto.MessageName(protobufResponse)
+		isExceptionPath := exceptionPaths.Has(pbQueryResponseTypeUrl)
+		splitResponse := strings.Split(pbQueryResponseTypeUrl, "Response")
+		assert.Lenf(t, splitResponse, 2, "typeUrl: %v",
+			splitResponse, pbQueryResponseTypeUrl)
+
+		// Get proto message "package" from the response type
+		typeUrlMinusSuffix := splitResponse[0]
+		typeUrlPartsFromProtoMsg := strings.Split(typeUrlMinusSuffix, ".")
+		assert.GreaterOrEqual(t, len(typeUrlPartsFromProtoMsg), 4, typeUrlPartsFromProtoMsg)
+		protoMessagePackage := typeUrlPartsFromProtoMsg[:3]
+
+		// Get proto message "package" from the query path
+		typeUrlPartsFromQueryPath := strings.Split(queryPath, ".")
+		assert.GreaterOrEqual(t, len(typeUrlPartsFromQueryPath), 4, typeUrlPartsFromQueryPath)
+		queryPathProtoPackage := typeUrlPartsFromQueryPath[:3]
+
+		// Verify that the packages match
+		assert.Equalf(t, queryPathProtoPackage, protoMessagePackage,
+			"package names inconsistent:\nfrom query path: %v\nfrom protobuf object: %v",
+			queryPath, pbQueryResponseTypeUrl,
+		)
+
+		// Verify that the method names match too.
+		methodNameFromPb := strings.TrimLeft(typeUrlPartsFromProtoMsg[3], "Query")
+		methodNameFromPath := strings.TrimLeft(typeUrlPartsFromQueryPath[3], "Query/")
+		if !isExceptionPath {
```

Comment:
This could be moved up to line 109
```go
if isExceptionPath {
    continue
}
```

Prefer the "[early return pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8)" over logic wrapped in branches.

Reply:
Yes, good idea. Addressed in 6f4b57767b63a222982e1e2bfa27b215e8d7c570